src/main/java/com/linecorp/armeria/common/SerializationFormat.java

```diff
@@ -62,6 +74,15 @@
     public static Set<SerializationFormat> ofThrift() {
         return THRIFT_FORMAT;
     }
+
+    /**
+     * Returns the serialization format corresponding to the passed in {@code mimeType}, or
+     * {@link Optional#empty} if the mimetype is not recognized. {@code null} is treated as an unknown
+     * mimetype.
+     */
+    public static Optional<SerializationFormat> fromMimeType(@Nullable String mimeType) {
+        return Optional.ofNullable(MIME_TYPE_TO_FORMAT.get(mimeType));
```

Comment:
Perhaps we should do some regex pattern matching instead? e.g. `;\s*protocol\s*=\s*([^ \t]+)` We could write a full parser or use Guava `MediaType`, but I guess regex should be enough for us at the moment.